pkg/chartutil/values.go

```diff
@@ -133,6 +147,106 @@ func ReadValuesFile(filename string) (Values, error) {
 	return ReadValues(data)
 }
 
+// ReadSchemaFile will parse a YAML file into a Schema.
+func ReadSchemaFile(filename string) (Schema, error) {
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return Schema{}, err
+	}
+	return ReadSchema(data)
+}
+
+// ValidateAgainstSchema checks that values does not violate the structure laid out in schema
+func ValidateAgainstSchema(values Values, schema Schema) error {
+	valuesJSON := convertToJSON(values)
+	schemaJSON := convertToJSON(schema)
+	schemaLoader := gojsonschema.NewStringLoader(string(schemaJSON))
+	valuesLoader := gojsonschema.NewStringLoader(string(valuesJSON))
+
+	result, err := gojsonschema.Validate(schemaLoader, valuesLoader)
+	if err != nil {
+		return err
+	}
+
+	if !result.Valid() {
+		var sb strings.Builder
+		sb.WriteString(".Values does not meet the specification of values.schema.yaml . see errors :\n")
```

Comment:
Can we change the error output to the following:

> values don't meet the specification of the schema:\n

We may want to change the filename or allow more than YAML in the future, so by changing the wording to be a little more generic we can protect ourselves in the future.

Reply:
Thanks :)

I'll definitely be adding documentation - just wanted to make sure the code was good before spending the time.

One last issue - the [GenerateSchema](https://github.com/helm/helm/pull/5350/files#diff-c508e0fda01b39eddfe1d6a4449fe685R184) function has been implemented, but so far nothing references it. In the 2/27 dev call, @mattfarina mentioned potentially adding a command for it, which I'm also happy to do. Does that sound like something that could be useful?

Reply:
I imagine there's two ways we could tackle this problem:

1. Generate a `values.schema.yaml` along with the rest of the chart in `helm create`. That could tie in really nicely with `GenerateSchema`.
2. Introduce a new command: `helm generate schema`

I really love the [generator model Ruby on Rails introduced](https://guides.rubyonrails.org/command_line.html#rails-generate) to generate boilerplate code for you. `helm generate` could be used to generate the schema based on the values file in the chart. We could also `helm generate template` for adding new templates to a chart, `helm generate helper` for helper functions, etc.

I think I'm getting a little ahead of myself though. I'm heading off on a tangent :smile: 

@mattfarina do you recall what command were you thinking of? How would the UX look?

Tentatively I'd say keep GenerateSchema for now. Other third-party projects may find it useful!